src/main/java/ladysnake/effective/client/world/WaterfallCloudGenerators.java

```diff
@@ -14,6 +15,14 @@
 public class WaterfallCloudGenerators {
     public static final List<WaterfallCloudGenerator> generators = new ArrayList<>();
 
+    public static void tryAddGenerator(BlockRenderView world, BlockPos pos) {
+        final BlockPos abovePos = pos.up();
+
+        if (world.getBlockState(pos).getBlock() == Blocks.WATER && world.getBlockState(pos).getFluidState().isStill() && world.getBlockState(abovePos).getBlock() == Blocks.WATER && !world.getBlockState(abovePos).getFluidState().isStill() && world.getBlockState(abovePos).getFluidState().getHeight() >= 0.77f) {
```

Comment:
May I suggest storing the blockstates in local variables to avoid extraneous getBlockState calls ?